File: frontend/src/types/delayPrediction.ts
export interface DelayPrediction {
  prediction_id: string;
  trip_id: string;
  delay_probability: number;
  is_delayed_prediction: boolean;
  model_version: string;
  features: Record<string, unknown>;
  predicted_at: string;
}

export interface ExpectedDelay {
  predicted_delay_minutes: number;
}

File: backend/app/schemas/delay_prediction.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class DelayPredictionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    prediction_id: uuid.UUID
    trip_id: str
    delay_probability: float
    is_delayed_prediction: bool
    model_version: str
    features: dict[str, Any]
    predicted_at: datetime


class ExpectedDelayRead(BaseModel):
    predicted_delay_minutes: float
